File: 1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder sb = new StringBuilder();

        int idx = 0;

        while(idx < word1.length() || idx < word2.length()){
            if(idx<word1.length())
                sb.append(word1.charAt(idx));
            
            if(idx<word2.length())
                sb.append(word2.charAt(idx));
            
            idx++;
        }
        
        return sb.toString();
    }
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        ans = []

        n, m = 0, 0

        while n < len(word1) or m < len(word2):
            if n < len(word1):
                ans.append(word1[n])
                n+=1
            
            if m < len(word2):
                ans.append(word2[m])
                m+=1
        
        return "".join(ans)
            
            